### ISSUE TITLE
Collision polygons in TileMap removed from all same objects

### ISSUE THREAD

**Issue by iRumba** [godotengine/godot#81904]
### Godot version

4.1.1.stable

### System information

Windows 10

### Issue description

I create TileMap level with platforms and coins.

Coins have collision polygon and some custom data (for identification as coins)

In code I enumerate all cells.

for other cell i check custom data and if it is coin I add to level Area2D to polygon placement and remove polygon with command
`tile_data.remove_collision_polygon(0, 0)`

But after this command all coins miss their polygon

I dont know is this bug or feature :)

If this is feature, please help me with replace physics polygons to Areas

### Steps to reproduce

1. Create project
2. Add TileMap to scene
3. Create one tile with physics polygon
4. Add several tiles to map
5. In code remove polygon from any tile
6. Check other tile for polygon

### Minimal reproduction project

None

**Comment by Zireael07** [comment]
IIRC collision polygons are resources, therefore shared by default.

**Comment by Mickeon** [comment]
Resources are most often shared. The coins they all share the same CollisionPolygon2D Resource by design. If they didn't, it would increase memory usage by a huge margin.

This could be a misunderstanding of the TileMap/TileSet. Have you tried using Scene Collections for your TileSet? Look for _scenes collection_ in this [admittedly verbose tutorial](https://docs.godotengine.org/en/stable/tutorials/2d/using_tilemaps.htm).
If you set up the "Coin" to be a Scene and make a Scene Collection with it, you may then place it on the TileMap as if it were a tile, aligned to the grid.

**Comment by groud** [comment]
This is expected behavior. TileData objects are part of a TileSet, which is a shared resource for all tiles. If you modify the TileData object directly, they you modify the TileData for all cells. 

To modify the TileData on a per-cell basis, a copy needs to be created, which is done by using the "runtime update". See https://docs.godotengine.org/en/stable/classes/class_tilemap.html#class-tilemap-method-tile-data-runtime-update

**Comment by iRumba** [comment]
> Resources are most often shared. The coins all share the same CollisionPolygon2D Resource by design

yes i understand. But why `tile_data.remove_collision_polygon(0, 0)` deletes Resource instead of resource link from the tile_data?
